### PR TITLE
enh(conf/broker) new unified_sql output in broker configuration

### PR DESCRIPTION
**Pull request by a-launois**
## Description

Deprecate broker outputs:
- SQL - Broker SQL database
- Storage - Perfdata Generator (Centreon Storage) 

Create broker output : 
- Storage - Unified SQL

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/install/php/Update-22.04.0-beta.1.php [hunk 1]
```diff
@@ -135,6 +135,87 @@
...
+    $errorMessage = 'Unable to update cb_type table ';
+
+    $statement = $pearDB->query(
+        "UPDATE `cb_type` set type_name = 'Perfdata Generator (Centreon Storage) - DEPRECATED'
+            WHERE type_shortname = 'storage'"
+    );
+    $statement = $pearDB->query(
+        "UPDATE `cb_type` set type_name = 'Broker SQL database - DEPRECATED'
+            WHERE type_shortname = 'sql'"
+    );
```
Comment (jeremyjaouen) [inline www/install/php/Update-22.04.0-beta.1.php:153]:
```suggestion
    $pearDB->query(
        "UPDATE `cb_type` set type_name = 'Perfdata Generator (Centreon Storage) - DEPRECATED'
            WHERE type_shortname = 'storage'"
    );
    $pearDB->query(
        "UPDATE `cb_type` set type_name = 'Broker SQL database - DEPRECATED'
            WHERE type_shortname = 'sql'"
    );
```
No need to store your query in a variable as you don't do anything with it

File: www/install/php/Update-22.04.0-beta.1.php [hunk 2]
```diff
@@ -135,6 +135,87 @@
...
+
+    $statement = $pearDB->query("SELECT cb_module_id FROM cb_module WHERE name = 'Storage'");
+    $module = $statement->fetch();
+
+    if ($module === false) {
+        throw new Exception("Cannot find 'Storage' module in cb_module table");
+    }
+    $moduleId = $module['cb_module_id'];
+
+    $statement = $pearDB->query(
```
Comment (jeremyjaouen) [inline www/install/php/Update-22.04.0-beta.1.php:163]:
```suggestion
    $pearDB->query(
```

File: www/install/php/Update-22.04.0-beta.1.php [hunk 3]
```diff
@@ -135,6 +135,87 @@
...
+    $errorMessage = 'Unable to update cb_tag_type_relation table';
+    $statement = $pearDB->query("SELECT cb_tag_id FROM cb_tag WHERE tagname = 'Output'");
+    $tag = $statement->fetch();
+
+    if ($tag === false) {
+        throw new Exception("Cannot find 'Output' tag in cb_tag table");
+    }
+    $tagId = $tag['cb_tag_id'];
+
+    $statement = $pearDB->query(
```
Comment (jeremyjaouen) [inline www/install/php/Update-22.04.0-beta.1.php:178]:
```suggestion
    $pearDB->query(
```

File: www/install/php/Update-22.04.0-beta.1.php [hunk 4]
```diff
@@ -135,6 +135,87 @@
...
+        "INSERT INTO `cb_tag_type_relation` (`cb_tag_id`, `cb_type_id`, `cb_type_uniq`)
+            VALUES ($tagId, $typeId, 0)"
+    );
+
+    $errorMessage = 'Unable to update options table';
+    $statement = $pearDB->query("INSERT INTO options VALUES ('unified_sql_db_type', 'mysql')");
```
Comment (jeremyjaouen) [inline www/install/php/Update-22.04.0-beta.1.php:184]:
```suggestion
    $pearDB->query("INSERT INTO options VALUES ('unified_sql_db_type', 'mysql')");
```

File: www/install/php/Update-22.04.0-beta.1.php [hunk 5]
```diff
@@ -135,6 +135,87 @@
...
+
+    $errorMessage = 'Unable to update cb_field table';
+    $statement = $pearDB->query(
```
Comment (jeremyjaouen) [inline www/install/php/Update-22.04.0-beta.1.php:187]:
```suggestion
    $pearDB->query(
```

**Review by jeremyjaouen — CHANGES_REQUESTED** [review]

File: www/install/php/Update-22.04.0-beta.1.php [hunk 6]
```diff
@@ -150,3 +170,73 @@
 
     throw new \Exception($versionOfTheUpgrade . $errorMessage, (int)$e->getCode(), $e);
 }
+
+/**
+ * Handle new broker output creation 'unified_sql'
+ *
+ * @param CentreonDB $pearDB
+ * @return void
```
Comment (jeremyjaouen) [inline www/install/php/Update-22.04.0-beta.1.php:178]:
```suggestion
```

**Review by jeremyjaouen — DISMISSED** [review]

**Review by jeremyjaouen — DISMISSED** [review]

**Review by kduret — DISMISSED** [review]

**Review by kduret — DISMISSED** [review]

**Review by kduret — DISMISSED** [review]

File: www/install/php/Update-22.04.0-beta.1.php [hunk 7]
```diff
@@ -150,3 +170,72 @@
...
+    }
+
+    $inputs[] = ['cb_field_id' => $fieldId, 'is_required' => 1];
+
+    $query = "INSERT INTO `cb_type_field_relation` (`cb_type_id`, `cb_field_id`, `is_required`, `order_display`)";
+    foreach ($inputs as $key => $input) {
+        $order = $key + 1;
+        $query .= $key === 0 ? " VALUES " : ", ";
+        $query .= "($typeId, " . $input['cb_field_id'] . ", " . $input['is_required'] . ", $order)";
+    }
```
Comment (callapa) [inline www/install/php/Update-22.04.0-beta.1.php:239]: Please to use the prepare method with the bindParam instead

**Review by callapa — CHANGES_REQUESTED** [review]

**Review by jeremyjaouen — DISMISSED** [review]

**Review by AssanDiomande — DISMISSED** [review]

**Review by kduret — APPROVED** [review]

**Review by AssanDiomande — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Pull request merged**